src/loop_util/loop_util.c

```diff
@@ -0,0 +1,137 @@
+/* Copyright (c) 2015 CoreOS. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+#include <sys/types.h>
+#include <sys/mount.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <linux/loop.h>
+#include <sys/file.h>
+#include <libmount/libmount.h>
+
+/*#include "path-util.h"
+#include "loopback-setup.h"
+#include "dev-setup.h"
+#include "fdset.h"*/
+
+
+static int get_node(const char *image_path, char *source) {
+
+        /*
+
+        struct loop_info64 info = {
+                .lo_flags = LO_FLAGS_AUTOCLEAR|LO_FLAGS_PARTSCAN
+        };
+
+        */
+
+        int fd = -1, nr = -1, loop = -1, control = -1;
+        
+        char* loopdev = NULL;  // free this pointer!
```

Comment:
obvious comment is not needed, *but* you don't actually free it. I would recommend do cleanup via goto, so something like:

```c
  int ret = -1;

  fd = open();
  if (fd < 0) {
    perror();
    goto out;
  }

  ret = ioctl();
  if (ret < 0) {
    perror();
    goto out;
  }

  /* everything else */

  ret = 0;

out:
  if (fd >= 0)
    close(fd);
  free(loopdev);
  return ret;
```